### PR TITLE
CS-30

### PR DESCRIPTION
**Pull request by Alvindelito**
### This Pull Request:

- Updates Header.js to accept props
- Updates CalendarView.js to reflect new changes to Header.js while still keeping the same style & functionality
- Adds ContactReview.js and ContactReviewStyles.js files to Scheduling folder
- Adds new route `/contactreview` to App.js that renders ContactReview.js

Notes: Edit button was positioned using `{position: 'absolute'}`. I had to set different CSS `bottom` values to position each button accordingly. It feels hacky, but this was the only solution I found so far.

### Screenshots
Contact Review 
Screen Shot 2021-05-02 at 9 18 02 PM

Contact Review where email and phone number are being edited
Screen Shot 2021-05-02 at 9 20 21 PM

### CODE REVIEW

File: app/src/components/Scheduling/ContactReview.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
+import React, { useEffect, useState } from 'react';
+import Header from './Header';
+import {
+  TextField,
+  FormControl,
+  Button,
+} from '../../store/index';
+import { useStyles } from './ContactReviewStyles';
+
+const HARD_CODED_DATA = {
```
Comment (jonathanbutler7) [inline app/src/components/Scheduling/ContactReview.js:10]: Good idea

File: app/src/components/Scheduling/ContactReview.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    email: false,
+    phoneNumber: false,
+  });
+
+  useEffect(() => {
+    setInputs(HARD_CODED_DATA);
+  }, [])
+
+  const onChange = (e) => {
+    setInputs({...inputs, [e.target.name]: e.target.value})
```
Comment (jonathanbutler7) [inline app/src/components/Scheduling/ContactReview.js:38]: I like the pattern of `{...inputs, [e.target.name]: e.target.value}` but I'd prefer calling this function `handleChange`, since this is what the react docs use: https://reactjs.org/docs/forms.html

File: app/src/components/Scheduling/ContactReview.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        title="Contact Information"
+        description="Review your profile information below"
+        activity={`Training with ${HARD_CODED_DATA.caseWorker}`}
+      />
+
+      <FormControl>
+        <>
+          <label className={classes.letterSpacing}>First Name<span className={classes.required}>*</span>
+          </label>
+          <TextField
```
Comment (jonathanbutler7) [inline app/src/components/Scheduling/ContactReview.js:53]: I spent a few minutes trying to figure out why these <inputs> get rendered with so much padding. In this mock, the inputs have a height of 48px: https://www.figma.com/file/kwUcXHYpKMHhuktdElLmmw/Calendaring-Home-Unite-Us-Copy?node-id=140%3A969

But in the screenshot, they're about 58px. I think it's because there's a lot of padding (circled in the screenshot), but I couldn't pinpoint where it's getting that padding from. I can look again tomorrow. 

Screen Shot 2021-05-02 at 22 25 59

File: app/src/components/Scheduling/ContactReview.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+            InputProps={{
+              classes: {
+                disabled: classes.disabled,
+                input: classes.inputWidth
+              },
+            }}
+            required
+          />
+          <Button
+            style={{bottom: '9rem'}}
```
Comment (jonathanbutler7) [inline app/src/components/Scheduling/ContactReview.js:105]: will take a closer look at this.

File: app/src/components/Scheduling/Header.js [hunk 5]
```diff
@@ -2,26 +2,14 @@ import React from 'react';
...
-      return 'Date selected';
-    }
-  };
-
   return (
     <div className={classes.root}>
       <div className={classes.header}>
         <img src={logo.default} className='logo' alt='logo' />
-        <Typography variant='h4'>Schedule with Tracy</Typography>
+        <Typography variant='h4'>{title}</Typography>
```
Comment (jonathanbutler7) [inline app/src/components/Scheduling/Header.js:12]: Great idea

**Review by jonathanbutler7 — COMMENTED** [review]
This is a great PR. Thank you for the detailed description and UI screenshots.

File: app/src/components/Scheduling/Header.js [hunk 6]
```diff
@@ -41,15 +27,14 @@ export default function Header({ textOptions }) {
...
       </Typography>
       <Typography
-        variant='body2'
+        variant='subtitle1'
         style={{
-          fontSize: '14px',
           marginBottom: '1rem',
         }}>
-        {getActionText()}
+        {children}
```
Comment (jonathanbutler7) [inline app/src/components/Scheduling/Header.js:37]: What should `children` be? When I log it to the console it's `undefined`. I also have doubts that children is the correct var name for this instance. Children is used more often a component is passed as props.

File: app/src/components/Scheduling/Header.js [hunk 7]
```diff
@@ -30,9 +18,7 @@ export default function Header({ textOptions }) {
           width: '200px',
           textAlign: 'center',
```
Comment (jonathanbutler7) [inline app/src/components/Scheduling/Header.js:19]: Line 26 of this component is commented out. If it's not being used, please remove it.

File: app/src/components/Scheduling/ContactReview.js (hunk 3 above)
Comment (jonathanbutler7) [reply app/src/components/Scheduling/ContactReview.js:53]: @Alvindelito Do you know where all this padding is coming from?

File: app/src/components/Scheduling/ContactReview.js [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+  return (
+    <div className={classes.root}>
+      <Header
+        title="Contact Information"
+        description="Review your profile information below"
+        activity={`Training with ${HARD_CODED_DATA.caseWorker}`}
+      />
+
+      <FormControl>
+        <>
```
Comment (jonathanbutler7) [inline app/src/components/Scheduling/ContactReview.js:50]: Curious what your thinking was behind using the fragments (`<></>`) within the return statement. They don't have any stylistic or semantic meaning, and they are not rendered to the page. https://reactjs.org/docs/fragments.html

Because this component's return statement is wrapped in a <div>, you the fragments are unneeded.

File: app/src/components/Scheduling/ContactReviewStyles.js [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+  },
+  editButton: {
+    width: '37px',
+    height: '22px',
+    position: 'absolute',
+    right: '0',
+    left: '16rem',
+    textTransform: 'capitalize',
+    letterSpacing: '1px',
+
```
Comment (jonathanbutler7) [inline app/src/components/Scheduling/ContactReviewStyles.js:29]: Extra line.

File: app/src/components/Scheduling/ContactReviewStyles.js [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+    alignItems: 'center',
+  },
+  formSpace: {
+    marginBottom: '1.5rem',
+  },
+  letterSpacing: {
+    letterSpacing: '1px',
+  },
+  editButton: {
+    width: '37px',
```
Comment (jonathanbutler7) [inline app/src/components/Scheduling/ContactReviewStyles.js:22]: Lines 22 and 23 can be removed.

File: app/src/components/Scheduling/ContactReview.js [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+            disabled={!editMode.phoneNumber}
+            InputProps={{
+              classes: {
+                disabled: classes.disabled,
+                input: classes.inputWidth
+              },
+            }}
+          />
+          <Button
+            style={{bottom: '2.7rem'}}
```
Comment (jonathanbutler7) [inline app/src/components/Scheduling/ContactReview.js:134]: Remove this line so we can keep all styles in one place. Same for line 105, and a few other places in different components.

File: app/src/components/Scheduling/ContactReviewStyles.js [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+  root: {
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'center',
+    margin: '1rem',
+  },
+  input: {
+    display: 'inline-flex',
+    flexDirection: 'row',
+    alignItems: 'center',
```
Comment (jonathanbutler7) [inline app/src/components/Scheduling/ContactReviewStyles.js:13]: To address your 'hacky' solution, add `position: relative;` here, and 

```js
right: 0,
top: 13,
```
To the editButton class on line 22. (It's a javascript object so you can enter integer values as opposed to `right: '0px';`)

File: app/src/components/Scheduling/ContactReviewStyles.js [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+  buttonContainer: {
+    position: 'absolute',
+    bottom: 0,
+    right: 0,
+    margin: '1rem',
+  },
+  inputWidth: {
+    width: '65%',
+  },
+  required: {
```
Comment (jonathanbutler7) [inline app/src/components/Scheduling/ContactReviewStyles.js:40]: Might be better to add a `warn` color to the theme object. If you want to do this, that's fine, but it's not really part of this PR...I may restructure the project a little this weekend anyway.

**Review by jonathanbutler7 — CHANGES_REQUESTED** [review]
Great work. Hope you don't mind me being nit-picky again.

File: app/src/components/Scheduling/Header.js (hunk 6 above)
Comment (Alvindelito) [reply app/src/components/Scheduling/Header.js:37]: Ahh I see. I definitely named it wrong then. `children` renders the action text depending on the page. These are seen in the calendar view and choose time availability pages. 

I'm thinking the content in `children` can be moved to the page itself (CalendarView) and placed right below the `<Header />`. Then remove the `children` prop from `<Header />`.

This is how I would restructure the header in `CalendarView.js` for the above solution:

```
<Header
  title="Schedule with Tracy"
  description="Select from Tracy's availability for"
  activity="Training"
>
  {buttonDisabled
    ? 'Choose a date'
    : 'Date selected'
  }
</Header>
```

To:

```
<Header
  title="Schedule with Tracy"
  description="Select from Tracy's availability for"
  activity="Training"
/>
<h5>
  {buttonDisabled
    ? 'Choose a date'
    : 'Date selected'
  }
</h5>
```

A different soltuion I can think of is renaming `children` to `action`.

I'm currently leaning more toward the first solution to avoid having `children` be undefined when it is not being used in other pages such as the `<ContactReview />` page.

In this screenshot, it's the very bottom text that says "3 out of 3 days chosen".
Screen Shot 2021-05-04 at 8 50 41 AM

File: app/src/components/Scheduling/ContactReview.js (hunk 3 above)
Comment (Alvindelito) [reply app/src/components/Scheduling/ContactReview.js:53]: I'm not entirely sure how you got that extra padding. There shouldn't be an extra input in that space.

Here is what it looks like on my screen when after a fresh reload:
Screen Shot 2021-05-04 at 9 20 50 AM

The extra space between `Training with Tracy`  and `First Name*` is coming from flexbox

File: app/src/components/Scheduling/ContactReview.js (hunk 8 above)
Comment (Alvindelito) [reply app/src/components/Scheduling/ContactReview.js:50]: I most definitely used fragments incorrectly. My thinking at the time was to separate them. But that was entirely wrong after looking at the docs. I removed the fragments.

File: app/src/components/Scheduling/ContactReview.js (hunk 3 above)
Comment (Alvindelito) [reply app/src/components/Scheduling/ContactReview.js:53]: Actually, after taking a closer look, I can see an extra `<h6>` in the space between `Training with Tracy` and `First Name*`. This is coming from `<Header />` and the extra space should go away if I get the okay to remove the `children` prop in the other comment

Comment (jonathanbutler7) [reply app/src/components/Scheduling/ContactReview.js:53]: My bad, that extra input was there because I was experimenting with something. I'm talking about the padding in the input fields, like the one that contains `Nina`. They're much taller than the 48px height in the Figma link in this thread.

Comment (jonathanbutler7) [reply app/src/components/Scheduling/ContactReview.js:53]: Good find on the extra h6 though.

File: app/src/components/Scheduling/ContactReviewStyles.js (hunk 12 above)
Comment (Alvindelito) [reply app/src/components/Scheduling/ContactReviewStyles.js:14]: Thank you so much!!

File: app/src/components/Scheduling/ContactReview.js (hunk 3 above)
Comment (Alvindelito) [reply app/src/components/Scheduling/ContactReview.js]: Ahhh I see now. This is the default style of `<TextField>`. I should've looked at it closer to match the Figma style. I'll get that fixed shortly

File: app/src/components/Scheduling/Header.js (hunk 6 above)
Comment (Alvindelito) [reply app/src/components/Scheduling/Header.js]: Just pushed the first solution!

**Review by jonathanbutler7 — APPROVED** [review]
Looks great
